core/src/main/scala/org/apache/spark/MapOutputTracker.scala

```diff
@@ -138,13 +138,17 @@ private[spark] abstract class MapOutputTracker(conf: SparkConf) extends Logging
    *         and the second item is a sequence of (shuffle block id, shuffle block size) tuples
    *         describing the shuffle blocks that are stored at that block manager.
    */
-  def getMapSizesByExecutorId(shuffleId: Int, startPartition: Int, endPartition: Int)
+  def getMapSizesByExecutorId(shuffleId: Int, startPartition: Int, endPartition: Int,
+                              startMapId: Option[Int] = None,
```

Comment:
If use `startMapId` and `endMapId`, it's too difficulty for the partition to be aware of the localtion. I suggest use `Seq[Int]` instead. Since the `Seq[Int]` will have a big size in closure, you can `bitmap` also.